### ISSUE TITLE
Ratelimiter Error during clean up

### ISSUE THREAD

**Issue by renja-g** [iann838/pulsefire#7]
I have a project that is constantly making requests using pulsefire and its ratelimiter, I just looked into the logs of the ratelimiter, and it seems like in very rare cases it throws an error like once every few days.
```
Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pulsefire/ratelimiters.py", line 195, in synchronize
    await self.synchronize(Invocation(**data["invocation"]), data["headers"])
  File "/usr/local/lib/python3.12/site-packages/pulsefire/ratelimiters.py", line 137, in synchronize
    for prev_uid, (prev_request_time, _) in self._track_syncs.items():
RuntimeError: dictionary changed size during iteration
```

The problem seems to be poping from the dict that is currently iterated over:
```py
        if random.random() < 0.1:
            for prev_uid in list(self._track_syncs.keys()):
                prev_request_time, _ = self._track_syncs[prev_uid]
                if response_time - prev_request_time > 600:
                    self._track_syncs.pop(prev_uid, None) # <-- This
``` 
[line 139](https://github.com/iann838/pulsefire/blob/master/pulsefire/ratelimiters.py#L139)

If you want to see how I use pulsefire here is the [repo](https://github.com/renja-g/Realm-Warp)

**Comment by iann838** [comment]
@renja-g Will send a fix shortly.

**Comment by iann838** [comment]
@renja-g fixed with v2.0.10, reopen if issue persist
